client/src/main/java/io/pravega/client/tables/TableTransaction.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import com.google.common.annotations.Beta;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines all operations that can be made on a Table Transaction.
+ *
+ * Notes about Updates:
+ * * Updates (Blind or Conditional) in this TableTransaction will not be reflected in the main Table until {@link #commit()}
+ * is invoked (and completes successfully).
+ * * Conditional Updates in this TableTransaction can only be done by comparing with Keys in this Transaction. Comparisons
+ * cannot be made against Keys from the main Table.
+ * * {@link #commit()} cannot be made conditional against the state of the main Table.
+ *
+ * @param <KeyT>   Table Key Type.
+ * @param <ValueT> Table Value Type.
+ */
+@Beta
+public interface TableTransaction<KeyT, ValueT> extends AutoCloseable {
```

Comment:
I understand this API and the contract it is trying to gurantee, but I'd like to understand a bit better how this is going to be implemented. Could you provide a high-level description, please?

Reply:
The PDP explains how this will work. In a nutshell, just like you have transactions for streams, where such transactions are made of a set of segments associated with the stream's segments, so too will we have transaction table segments associated with the table's segments. The merge process will be similar from the client's POV, with a bit more work needing to be done server-side.